### PR TITLE
Support loglikelihood mechanism

### PR DESCRIPTION
**Pull request by vvchernov**
Loglikelihood approach is needed to check accuracy of a model on popular datasets and tasks like MMLU and BigBench. Particularly HuggingFace leaderboard bases on such tasks only.

**Notes:**
1. Log softmax calculation is transferred onto TVM side. It should be up-streamed before merge. Just now the branch with method implementation is started from tvm/unity, but current 3d-party tvm is gotten from mlc-ai/relax. They have some conflicts and need additional efforts for upstreaming. Need to discuss.
2. Loglikelihood requires sequence of logits after prefill-like inference. But in mlc-llm on the model topology side the last logits set is splitted from all. I've skipped split op on topology side for all models excluding rwkv (need help there). Split has been implemented in mlc-llm

**TEST:**
Tests were performed with lm-evaluation-harness (master branch) on samples from Hellaswag task and Llama2-7b from HF and on mlc-llm side with on Llama2-7b-q0f16. Logprobes for context and four continuation:
Lm-evaluation-harness:
"logit_0": -83.95526123046875,
"logit_1": -40.397254943847656,
"logit_2": -48.10240173339844,
"logit_3": -70.90489196777344,
Mlc-llm:
"logit_0": -84.0309,
"logit_1": -40.476,
"logit_2": -48.3609,
"logit_3": -70.8229,

### CODE REVIEW

**Comment by vvchernov** [comment]
Hello @tqchen! I need your help and expertise to discuss how to better implement things described in notes. My quick understanding: 1. Transfer my work branch from tvm/unity to mlc-ai/relax; 2. allow the topology return all logits and cut the last one on chat side.

**Comment by tqchen** [comment]
mlc/relax is doing weekly sync with tvm/unity, so no actions needed. we are looking into a possible regression, after that is cleared, it should be updated

**Comment by vvchernov** [comment]
Close due to development was transferred to octoml/mlc-llm
